.github/scripts/trymerge.py

```diff
@@ -1265,6 +1295,90 @@ def checks_to_str(checks: List[Tuple[str, Optional[str]]]) -> str:
 def checks_to_markdown_bullets(checks: List[Tuple[str, Optional[str]]]) -> List[str]:
     return [f"- [{c[0]}]({c[1]})" if c[1] is not None else f"- {c[0]}" for c in checks[:5]]
 
+def get_flaky_rules() -> List[FlakyRule]:
+    url = "https://raw.githubusercontent.com/pytorch/test-infra/generated-stats/stats/flaky-rules.json"
+    flaky_rules_json: List[FlakyRule] = []
```

Comment:
As this is python code, you could try https://pypi.org/project/retry if you think it's worth the effort to install a new package.  That would help with other functions like `get_rockset_results` too.

Also if you don't want to install that package, this function could be simplify a bit to avoid `flaky_rules_json` getting into a weird state between retries

```
for _ in range(3):
    try:
        return [FlakyRule(**rule) for rule in fetch_json_list(url)]
    except Exception as e:
        print(f"Could not download {url} because: {e}.")
return []
```

Reply:
went with the code suggestion